system/src/system_update.cpp

```diff
@@ -318,7 +318,12 @@ int Spark_Finish_Firmware_Update(FileTransfer::Descriptor& file, uint32_t flags,
 
     hal_module_t mod;
 
-    if (flags & 1) {    // update successful
+    if ((flags & (UPDATE_FLAG_VALIDATE_ONLY | UPDATE_FLAG_SUCCESS)) == (UPDATE_FLAG_VALIDATE_ONLY | UPDATE_FLAG_SUCCESS)) {
```

Comment:
`(UPDATE_FLAG_VALIDATE_ONLY | UPDATE_FLAG_SUCCESS) != 1`. The first `if` branches us into a validate-only case when both `UPDATE_FLAG_VALIDATE_ONLY` and `UPDATE_FLAG_SUCCESS` are set. The one after is the original `flags & 1` which is equivalent to `flags & UPDATE_FLAG_SUCCESS`.